tests/cross_functional/scale/noobaa/test_delete_objects.py

```diff
@@ -190,3 +202,120 @@ def test_delete_objects(
+        self,
+        request,
+        reduce_expiration_interval_session,
+        change_lifecycle_schedule_min_session,
+        bucket_factory_session,
+        mcg_obj_session,
+    ):
+        buckets = list()
+        expirations = list()
+
+        def factory(exp):
+            """
+            Factory function to create the bucket
+
+            """
+
+            # Reduce the expiration interval and lifecycle schedule delay
+            log.info(
+                "Reducing the expiration interval and lifecycle schedule delay to 1 minute"
+            )
+            reduce_expiration_interval_session(interval=1)
+            change_lifecycle_schedule_min_session(interval=1)
+
+            expirations.append(exp)
+            bucket = bucket_factory_session(amount=1, interface="OC")[0]
+            buckets.append(bucket)
+            return bucket
+
+        def teardown():
+            """
+            Teardown function to verify the object deletion
+            in the bucket
+
+            """
+            for bucket, expiration in zip(buckets, expirations):
+                verify_objs_deleted_from_objmds(bucket.name, timeout=30000, sleep=60)
+                if expiration:
+                    sample_if_objects_expired(
+                        mcg_obj_session, bucket.name, timeout=36000, sleep=60
+                    )
+                log.info("Verified that all objects are deleted and marked deleted")
+
+        request.addfinalizer(teardown)
+        return factory
+
+    @scale_long_run
+    @jira("DFBUGS-1106")
+    @jira("DFBUGS-1116")
+    @pytest.mark.parametrize(
+        argnames=["expiration"],
+        argvalues=[
+            pytest.param(
+                True,
+                marks=[
+                    pytest.mark.jira("DFBUGS-1116"),
+                    pytest.mark.polarion_id("OCS-6097"),
+                ],
+            ),
+            pytest.param(
+                False,
+                marks=[
+                    pytest.mark.jira("DFBUGS-1106"),
+                    pytest.mark.polarion_id("OCS-6096"),
+                ],
+            ),
+        ],
+    )
+    def test_delete_objs_by_expiration_and_recursive_deletion(
+        self,
+        create_bucket_verify_object_deletion,
+        mcg_obj_session,
+        awscli_pod_session,
+        test_directory_setup,
+        expiration,
+    ):
+        """
+        Test to verify object deletion through object mds when we delete or expire
+        objects in bucket at scale
+
+        """
+
+        # Create the bucket
+        log.info("Create the bucket")
+        bucket = create_bucket_verify_object_deletion(exp=expiration)
+
+        # Generate and upload objects to the
+        # bucket parallely
+        gen_empty_file_and_upload(
+            mcg_obj_session,
+            awscli_pod_session,
+            test_directory_setup.origin_dir,
+            amount=500000,
+            bucket=bucket.name,
+            threads=10,
+            timeout=7200,
```

Comment:
1. Regarding Timeout --> are we expecting the expiration will complete after 120hrs which is 5 days? or How about using timeout_sampler to iteratively check if the object policy is expired?
2. Is the expiration time is configurable for object? if so could we simulate with less expiration time?

Reply:
@ramkiperiy 
1. So we have two parametrisation, hence test executes 2 times. Total execution time was about 10 hours. This includes object expiration verification taking 70% time which is happening at the end of the session scoped fixture. so it wont block other scale tests from executing. 
2.  It is only configurable for bucket. So we have configured 1day (it cant be less than this) but we are tweaking the noobaa background worker configuration to fasten this. Hence object expiration cleanup starts within few minutes.

Reply:
1. So even with expiration starts in few minutes the reason for long execution time is due to no of objects? If so TC expectation is to create lot of objs?
2. 1day expiration policy in bucket --> Any specific reason we can't reduce this?
Would like to understand if there is any option to optimise the test run time.